deposit_contract.sol

```diff
@@ -12,15 +13,25 @@ interface IDepositContract {
         bytes index
     );
 
+    /// @notice Submit a Phase 0 DepositData object.
```

Comment:
I think the comments have to be on the contract itself to be visible in the metadata. Docstring inheritance is still an open issue.

Reply:
I did check it as I was writing it (the only expected part is that events are ignored, see https://github.com/ethereum/solidity/issues/8776):

```json
User Documentation
{
  "methods":
  {
    "deposit(bytes,bytes,bytes,bytes32)":
    {
      "notice": "Submit a Phase 0 DepositData object."
    },
    "get_deposit_count()":
    {
      "notice": "Query the current deposit count."
    },
    "get_deposit_root()":
    {
      "notice": "Query the current deposit root hash."
    }
  }
}
```

```json
Developer Documentation
{
  "methods":
  {
    "deposit(bytes,bytes,bytes,bytes32)":
    {
      "params":
      {
        "deposit_data_root": "The SHA-256 hash of the SSZ-encoded DepositData object. Used as a protection against malformed input.",
        "pubkey": "A BLS12-381 public key.",
        "signature": "A BLS12-381 signature.",
        "withdrawal_credentials": "Commitment to a public key for withdrawals."
      }
    }
  }
}
```

Reply:
>I think the comments have to be on the contract itself to be visible in the metadata. Docstring inheritance is still an open issue.

Ah, understood after re-reading. Yes, what I pasted above is on the interface and not the contract, so it won't be included in the metadata.